src/main/scala/org/apache/spark/sql/delta/storage/S3SingleDriverLogStore.scala

```diff
@@ -157,14 +157,12 @@ class S3SingleDriverLogStore(
   override def write(path: Path, actions: Iterator[String], overwrite: Boolean = false): Unit = {
     val (fs, resolvedPath) = resolved(path)
     val lockedPath = getPathKey(resolvedPath)
-    var stream: FSDataOutputStream = null
     acquirePathLock(lockedPath)
     try {
       if (exists(fs, resolvedPath) && !overwrite) {
         throw new java.nio.file.FileAlreadyExistsException(resolvedPath.toUri.toString)
       }
-      val countingStream = new CountingOutputStream(stream)
-      stream = fs.create(resolvedPath, overwrite)
+      val stream = new CountingOutputStream(fs.create(resolvedPath, overwrite))
       actions.map(_ + "\n").map(_.getBytes(UTF_8)).foreach(stream.write)
       stream.close()
 
```

Comment:
I'm not 100% confident that CountingOutputStream will close the underlying stream. This way is just nice because there is no second handle laying around to get used by accident.

Reply:
Looking at https://commons.apache.org/proper/commons-io/javadocs/api-2.4/org/apache/commons/io/output/CountingOutputStream.html, it looks like it extends ProxyOutputStream, which provides the implementation of `.close()` and delegates it to `out`.